### ISSUE TITLE
Clearfix Compatibility

### ISSUE THREAD

**Issue by kmctown** [styled-components/polished#140]
This is a great library and I'm using it with `styled-components`, but having an issue `clearFix` interoping with bootstrap. The `clearFix` implementation is slightly different than bootstrap's and it doesn't seem to work with the Grid.

E.g., the `polished` implementation:

```css
&:after {
  clear: both;
  content: '';
  display: table;
}
```

vs. the Bootstrap implementation:

```css
&:before,
&:after {
  clear: both;
  display: table;
  content: "";
  line-height: 0;
}
```

Would be more than happy to submit a PR for this if you all are open to the change. Thanks!

**Comment by nikgraf** [comment]
Would be fine by me. @bhough @mxstbr any objections?

**Comment by bhough** [comment]
Our implementation is the accepted approach for all browsers (IE8+) and uses the minimum amount of properties and selectors to reduce potential impact on pseudo elements. I'm not sure I like the idea of of expanding that to accommodate one specific (albeit popular) library.

**Comment by kmctown** [comment]
Thanks for the feedback. Closing this...something else weird is going on. I created the following function in a util lib:

```
export function clearFix() {
  return `
    &:after {
      display: table;
      content: "";
      clear: both;
    }
  `;
}
```

And dropped it into styled components the same way:

```
const El = styled.div`
...
${clearFix()}
...
`
```

And it works, where the `clearFix` from polished doesn't. Does that make any sense?

**Comment by bhough** [comment]
Looks like the difference between the one we use and your example above is we use `&::after`  (double colon) where you are using a single colon. https://developer.mozilla.org/en-US/docs/Web/CSS/::after

Could be some browsers aren't liking the `::` colon version and we need to move to using the CSS2 version.

**Comment by kmctown** [comment]
Right, I tried it both ways. The `::` colon version works as well -- as long as I'm returning the string.

```
export function clearFix() {
  return `
    &::after {
      display: table;
      content: "";
      clear: both;
    }
  `;
}
```

⬆️  That works as well. Swapping out to the `polished` version breaks it. For whatever reason, chrome is not liking `content: ''` when using the version from `polished`.

Here is a screenshot from Chrome devtools when using the function that returns the string:

![image](https://cloud.githubusercontent.com/assets/1482857/24718499/3e3aafa8-1a04-11e7-9911-1c6ee39e1924.png)

And here is what it looks like when using the polished version:

![image](https://cloud.githubusercontent.com/assets/1482857/24718567/75b740f4-1a04-11e7-830b-3f4db6eb6997.png)

**Comment by bhough** [comment]
This *may* be an issue with `styled-components` not handling the object version correctly. This is what I get as output from our version:

`{"&::after":{"clear":"both","content":"","display":"table"}}`

My guess is `styled-components` is not handling the empty string correctly. @mxstbr @geelen thoughts?

**Comment by mxstbr** [comment]
I don't think this is a bug in `styled-components` (let's double check what other CSS-in-JS libs do here) but we should do this instead:

```JS
{
  "&::after": {
    "clear": "both",
    // Notice: Quotes inside quotes
    "content": "''",
    "display": "table"
  }
}
```

I think that'd fix it?

**Comment by bhough** [comment]
Closed by #157

**Comment by nikgraf** [comment]
sweet 👍
